hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableSink.java

```diff
@@ -106,17 +106,26 @@ public SinkRuntimeProvider getSinkRuntimeProvider(Context context) {
       // bootstrap
       final DataStream<HoodieRecord> hoodieRecordDataStream =
           Pipelines.bootstrap(conf, rowType, dataStream, context.isBounded(), overwrite);
+
       // write pipeline
       pipeline = Pipelines.hoodieStreamWrite(conf, hoodieRecordDataStream);
-      // compaction
+
+      // insert cluster mode
+      if (OptionsResolver.isInsertClusterMode(conf)) {
+        return Pipelines.clean(conf, pipeline);
+      }
```

Comment:
Why we handle the insert cluster mode separately.

Reply:
Mainly considering enhancing code readability，the insert cluster mode was together with the compaction logic before. I think it's better to be independent.

Reply:
Can you check the Travis test failures.